lib/graphql/resolvers/spreadsheet.resolver.ts

```diff
@@ -0,0 +1,52 @@
+import { Query } from "type-graphql";
+import { injectable } from "tsyringe";
+import { readSpreadsheet } from "../utilities/spreadsheet/setup";
+import { isEqual } from "lodash";
+import SpreadsheetService from "../services/Spreadsheet.service";
+import { SpreadsheetOverviewRevenueType, SpreadsheetOverviewDivisionsType } from "../schemas/Spreadsheet";
+
+@injectable()
+export default class spreadsheetResolver { 
+    constructor( private spreadsheetService : SpreadsheetService ) {} 
+
+    @Query( () => [SpreadsheetOverviewRevenueType] )
+    async getSpreadsheetOverviewRevenueData() {
+
+        const revenueData = await readSpreadsheet( ["Overview!B2:B", "Overview!D2:D", "Overview!F2:F" ] ); 
+        
+        if (!revenueData) return [] 
+        
+        const propertyNames = [ ['itemName', 'string'], ['budgetAmount', 'float'], ['notes', 'string']]
+        const startIndex = 1 ;
+        const endIndex =  revenueData?.at(0)?.values?.findIndex( obj => isEqual( obj, ['Sub Category'] ) )   || -1 
```

Comment:
There's no need to add question mark after `revenueData` since you already checked empty/null/undefined in line 17.